### ISSUE TITLE
"Remove all the unused imports" doesn't detect item-like usage of imports correctly

### ISSUE THREAD

**Issue by crumblingstatue** [rust-lang/rust-analyzer#15413]
main.rs
```Rust
use sub1::Foo;

mod sub1;
mod sub2;

fn main() {
    sub2::do_something();
}

```

sub1.rs:
```Rust
#[derive(Debug)]
pub struct Foo;
```

sub2.rs:
```Rust
pub fn do_something() {
    let f = crate::Foo;
    dbg!(f);
}
```

The "remove all the unused imports" is suggested for `use sub1::Foo;` in main. If applied, it removes the import, invalidating the code in `sub2.rs` which tries to use `crate::Foo`, which is valid because imports are item-like, and can be imported as such.

This seems to only happen if the modules are in different files. The following code, which inlines all the modules does not cause this behavior. The assist is not suggested:```Rust
use sub1::Foo;

mod sub1 {
    #[derive(Debug)]
    pub struct Foo;
}
mod sub2 {
    pub fn do_something() {
        let f = crate::Foo;
        dbg!(f);
    }
}

fn main() {
    sub2::do_something();
}
```

rust-analyzer version 0.3.1615-standalone

rustc version: rustc 1.73.0-nightly (04abc370b 2023-07-28)
